src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -122,14 +122,21 @@ def get_disk(self, name):
     @private
     def get_rotational_rate(self, device_path):
 
+        host_type = self.middleware.call_sync('system.dmidecode_info')['system-manufacturer']
+
         try:
             disk = libsgio.SCSIDevice(device_path)
             rotation_rate = disk.rotation_rate()
-        except Exception as e:
-            self.logger.error(
-                'Failed to retrieve rotational rate '
-                'for disk %s with error: %r', device_path, e
-            )
+        except RuntimeError:
+            # RuntimeError() means the ioctl failed
+            if host_type == 'QEMU':
+                # this is expected behavior on qemu/kvm guests
+                return
+            else:
+                self.logger.warning('Ioctl failed for %s', device_path)
+                return
+        except Exception:
```

Comment:
Now that we are logging this, how about having it here `(Exception, RuntimeError)` and the same logging would work with the if condition for host type plus exception type ? We avoid duplication this way